cli2/src/Morphir/Elm/CLI.elm

```diff
@@ -0,0 +1,175 @@
+
+
+update : Msg -> () -> ( (), Cmd Msg )
+update msg model =
+    case msg of
+        PackageDefinitionFromSource ( optionsJson, packageInfoJson, sourceFiles ) ->
+            let
+                inputResult : Result Decode.Error ( Frontend.Options, PackageInfo )
+                inputResult =
+                    Result.map2 Tuple.pair
+                        (Decode.decodeValue FrontendCodec.decodeOptions optionsJson)
+                        (Decode.decodeValue FrontendCodec.decodePackageInfo packageInfoJson)
+            in
+            case inputResult of
+                Ok ( _, packageInfo ) ->
+                    let
+                        emptyRepo : Repo
+                        emptyRepo =
+                            Repo.empty packageInfo.name
+                    in
+                    -- add source file processing
+                    ( model
+                    , emptyRepo
+                        |> Repo.toDistribution
+                        |> DistroCodec.encodeDistribution
+                        |> packageDefinitionFromSourceResult
+                    )
+
+                Err errorMessage ->
+                    ( model
+                    , errorMessage
+                        |> Decode.errorToString
+                        |> jsonDecodeError
+                    )
+
+        IncrementalBuild ( optionsJson, packageInfoJson, fileChangesJson, distribution ) ->
+            let
+                fileChangesResult : Result Decode.Error FileChanges
+                fileChangesResult =
+                    Decode.decodeValue FileChangesCodec.decodeFileChanges fileChangesJson
+
+                inputResult : Result Decode.Error ( Frontend.Options, PackageInfo )
+                inputResult =
+                    Result.map2 Tuple.pair
+                        (Decode.decodeValue FrontendCodec.decodeOptions optionsJson)
+                        (Decode.decodeValue FrontendCodec.decodePackageInfo packageInfoJson)
+
+                repoFromDistribution : Result Repo.Errors Repo
+                repoFromDistribution =
+                    Repo.fromDistribution distribution
+
+                result : ( (), Cmd msg )
+                result =
+                    case repoFromDistribution of
+                        Ok repo ->
+                            case fileChangesResult of
+                                Ok fileChanges ->
+                                    ( model
+                                    , Repo.applyFileChanges fileChanges repo
+                                        |> Result.map Repo.toDistribution
+                                        |> Result.mapError
+                                            (List.map
+                                                (\error ->
+                                                    case error of
+                                                        _ ->
+                                                            Compiler.ErrorsInSourceFile "" []
```

Comment:
Yes you are right

Reply:
When you have unfinished code it's better to use `Debug.todo` because a valid implementation like the above is easy to miss and can make it to production.

Reply:
Okay, noted